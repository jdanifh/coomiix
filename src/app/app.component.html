<cmx-layout [title]="title">
  <cmx-breadcrumbs header></cmx-breadcrumbs>
  <cmx-locale header></cmx-locale>
  <cmx-side-menu menu></cmx-side-menu>
  <router-outlet content></router-outlet>
</cmx-layout>
