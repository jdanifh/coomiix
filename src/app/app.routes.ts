import { Routes } from '@angular/router';

export const routes: Routes = [
  {
    path: '**',
    title: 'Home',
    loadComponent: () => import('./views/home/home.component').then(mod => mod.HomeComponent)
  },
];
