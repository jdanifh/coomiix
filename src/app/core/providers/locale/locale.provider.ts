import { importProvidersFrom, LOCALE_ID } from '@angular/core';
import { HttpBackend, HttpClient } from '@angular/common/http';
import { MAT_DATE_LOCALE } from '@angular/material/core';
import { MAT_MOMENT_DATE_FORMATS, provideMomentDateAdapter } from '@angular/material-moment-adapter';
import {TranslateModule, TranslateLoader} from "@ngx-translate/core";
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { registerLocaleData } from '@angular/common';

// EN Locale
import en from '@angular/common/locales/en';
registerLocaleData(en);
// ES Locale
import 'moment/locale/es';
import es from '@angular/common/locales/es';
registerLocaleData(es);

export enum LOCALES {
  EN = 'en-US',
  ES = 'es-ES'
}
export const DEFAULT_LOCALE = LOCALES.EN;

export const provideLocales = () => {
  return [
    {provide: LOCALE_ID, useValue: DEFAULT_LOCALE},
    {provide: MAT_DATE_LOCALE, useValue: DEFAULT_LOCALE},
    provideMomentDateAdapter(MAT_MOMENT_DATE_FORMATS, {useUtc: true}),
    importProvidersFrom([TranslateModule.forRoot({
      defaultLanguage: DEFAULT_LOCALE,
      loader: {
        provide: TranslateLoader,
        deps: [HttpBackend],
        useFactory: (http: HttpBackend) => new TranslateHttpLoader(new HttpClient(http), './i18n/', '.json'),
      },
    })])
  ]
};

export const provideLocalesTest = () => {
  return [
    provideLocales(),
    { provide: HttpBackend, useClass: HttpBackend },
  ]
}
