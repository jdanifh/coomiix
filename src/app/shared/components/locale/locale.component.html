<button [matMenuTriggerFor]="menu" mat-icon-button aria-label="change locale">
  <mat-icon aria-hidden="true">language</mat-icon>
</button>
<mat-menu #menu="matMenu">
  @for (locale of locales; track locale) {
    <button (click)="changeLocale(locale)" mat-menu-item>
      <span class="locale">{{ locale | uppercase | slice:0:2 }}</span>
      <span>{{ locale }}</span>
    </button>
  }
</mat-menu>
