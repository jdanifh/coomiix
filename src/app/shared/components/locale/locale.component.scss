:host {
  margin-right: 25px;
  margin-left: auto;
}

.locale {
  font-weight: 700;
  margin-right: 25px;
}
