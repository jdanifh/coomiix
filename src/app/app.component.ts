import { Component, ChangeDetectionStrategy } from '@angular/core';
import { RouterOutlet } from '@angular/router';
import { LayoutComponent } from "./shared/components/layout/layout.component";
import { SideMenuComponent } from "./shared/components/side-menu/side-menu.component";
import { LocaleComponent } from "./shared/components/locale/locale.component";
import { BreadcrumbsComponent } from "./shared/components/breadcrumbs/breadcrumbs.component";

@Component({
  selector: 'app-root',
  standalone: true,
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [
    RouterOutlet,
    LayoutComponent,
    SideMenuComponent,
    LocaleComponent,
    BreadcrumbsComponent
],
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss'
})
export class AppComponent {
  public title = 'Coomiix';
}
