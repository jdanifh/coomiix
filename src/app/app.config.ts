import { ApplicationConfig, provideExperimentalZonelessChangeDetection, provideZoneChangeDetection } from '@angular/core';
import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
import { provideRouter, withDebugTracing, withInMemoryScrolling, withViewTransitions } from '@angular/router';
import { provideHttpClient, withInterceptors } from '@angular/common/http';
import { provideLocales } from './core/providers/locale/locale.provider';

import { routes } from './app.routes';

// Zone.js is disabled in favor of zoneless change detection
export const appConfig: ApplicationConfig = {
  providers: [
    provideExperimentalZonelessChangeDetection(),
    provideAnimationsAsync(),
    provideRouter(routes, withDebugTracing(), withViewTransitions(), withInMemoryScrolling()),
    provideHttpClient(withInterceptors([])),
    provideLocales(),
  ]
};
