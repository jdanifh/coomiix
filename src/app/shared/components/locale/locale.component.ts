import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatMenuModule } from '@angular/material/menu';
import { LocaleService } from './locale.service';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'cmx-locale',
  imports: [
    CommonModule,
    MatIconModule,
    MatButtonModule,
    MatMenuModule,
  ],
  templateUrl: './locale.component.html',
  styleUrl: './locale.component.scss',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class LocaleComponent {

  private _service = inject(LocaleService);

  public locale = this._service.locale;
  public locales = this._service.locales;

  public changeLocale(locale: string) {
    this._service.changeLocale(locale);
  }

}
