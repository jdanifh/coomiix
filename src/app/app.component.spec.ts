import { TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { provideRouter } from '@angular/router';
import { routes } from './app.routes';
import { By } from '@angular/platform-browser';
import { provideLocales } from './core/providers/locale/locale.provider';
import { HttpBackend } from '@angular/common/http';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent],
      providers: [
        provideRouter(routes),
        provideLocales(),
        { provide: HttpBackend, useClass: HttpBackend },
      ]
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have the 'coomiix' title`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('Coomiix');
  });

  it(`should match snapshot`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();

    expect(fixture.nativeElement.innerHTML).toMatchSnapshot();
  })

});
