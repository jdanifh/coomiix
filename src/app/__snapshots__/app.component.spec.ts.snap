// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `"<cmx-layout ng-reflect-title="Coomiix"><header><a routerlink="/" matripple="" class="mat-ripple title" ng-reflect-router-link="/" href="/"><span>Coomiix</span></a><button mat-icon-button="" aria-label="toggle menu" mat-ripple-loader-uninitialized="" mat-ripple-loader-class-name="mat-mdc-button-ripple" class="mdc-icon-button mat-mdc-icon-button mat-unthemed mat-mdc-button-base" mat-ripple-loader-centered=""><span class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"></span><mat-icon role="img" aria-hidden="true" class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color" data-mat-icon-type="font">more_vert</mat-icon><span class="mat-focus-indicator"></span><span class="mat-mdc-button-touch-target"></span></button><cmx-breadcrumbs header=""><!--container--></cmx-breadcrumbs><cmx-locale header=""><button mat-icon-button="" aria-label="change locale" class="mat-mdc-menu-trigger mdc-icon-button mat-mdc-icon-button mat-unthemed mat-mdc-button-base" mat-ripple-loader-uninitialized="" mat-ripple-loader-class-name="mat-mdc-button-ripple" mat-ripple-loader-centered="" ng-reflect-menu="[object Object]" aria-haspopup="menu" aria-expanded="false"><span class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"></span><mat-icon role="img" aria-hidden="true" class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color" data-mat-icon-type="font">language</mat-icon><span class="mat-focus-indicator"></span><span class="mat-mdc-button-touch-target"></span></button><!--container--><mat-menu><!--container--></mat-menu></cmx-locale></header><aside><!--container--><cmx-side-menu menu=""><a routerlink="/" routerlinkactive="active" matripple="" class="mat-ripple" ng-reflect-router-link="/" ng-reflect-router-link-active="active" href="/"><mat-icon role="img" aria-hidden="true" class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color" data-mat-icon-type="font">home</mat-icon><span>cmx-side-menu.home</span></a><a routerlink="/dashboard" routerlinkactive="active" matripple="" class="mat-ripple" ng-reflect-router-link="/dashboard" ng-reflect-router-link-active="active" href="/dashboard"><mat-icon role="img" aria-hidden="true" class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color" data-mat-icon-type="font">dashboard</mat-icon><span>cmx-side-menu.dashboard</span></a></cmx-side-menu></aside><main><router-outlet content=""></router-outlet><!--container--></main></cmx-layout>"`;
